shell/browser/api/electron_api_menu_mac.mm

```diff
@@ -57,14 +62,19 @@
   if (!native_window)
     return;
 
+  base::WeakPtr<WebFrameMain> weak_frame;
+  if (frame) {
```

Comment:
This is slightly brittle, it's possible to give the optional a `nullptr` s.t. the optional is nonempty with a null pointer and this paragraph results in a nullptr dereference

Maybe replace `std::optional<WebFrameMain*>` with just a `WebFrameMain*` that may be nullptr?

Reply:
> Maybe replace std::optional<WebFrameMain*> with just a WebFrameMain* that may be nullptr?

I don't think the current gin converter will take `Undefined` and convert it to `nullptr` 🤔 Nor do I think it should.

Given the above I don't think it's possible in this case to have the optional wrap a `nullptr` as the converter would never put one there, but that isn't enforced by semantics so I do see the issue still.

Maybe the gin converter should go to `std::reference_wrapper<T>` instead of `T*` to indicate it's non-null nature, but thats a bit of a lift for the whole codebase

Reply:
Maybe an alternative could be to pass the [`content::GlobalRenderFrameHostToken`](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/render_frame_host.h;l=238-240;drc=f4daadec4cf496926ed23bb014749f7df81f6221) and retrieve the RenderFrameHost when needed. The token is used internally by WebFrameMain to lookup the RenderFrameHost. It's also recommended by the docs.

```cpp
// Returns the GlobalRenderFrameHostToken for this frame. Embedders should
// store this instead of a raw RenderFrameHost pointer.
virtual GlobalRenderFrameHostToken GetGlobalFrameToken() const = 0;
```

Reply:
@samuelmaddock This value comes from JS, do we have existing precedent for a `frameToken` being passed around 🤔

Reply:
Ah sorry, I misunderstood the initial discussion. The `GlobalRenderFrameHostToken` could be an alternative to passing the weak ptr. I'm okay with either though.

As for the argument converter, I believe you could instead accept a `content::RenderFrameHost*` and it'll be converted from a WebFrameMain for you.

https://github.com/electron/electron/blob/da8288106be98912bf58aba346e927d7395b1679/shell/common/gin_converters/frame_converter.cc#L37-L48

Reply:
I opted for a `foo && foo.value()` check to ensure no nullptr in this case, I think it would be good to consider a codebase wide change to something more declarative like `std::reference_wrapper` but that requires more thought / everything doing it so for now I'll just make this bit ok